### PR TITLE
Accessibility: It's own prefs section & Loud sound blocking.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Adds 'Accessibility' as its own section of the game preferences UI menu
Also adds a new Accessibility option, to block loud noises. This is used for Gravity Generator and Telecomms, whose sounds can easily disturb people with mental conditions.

what it looks like in-game:
image

## Why It's Good For The Game

It's helpful for people who need it, since it is ear-breaking and CONSTANT, especially if you're an Engineer working near one of these rooms.

## Changelog

:cl:
qol: Telecomms hubs are now the only telecomms machine that makes noise, lowering the amount of ear-breaking sound from being in the telecomms room.
add: Adds a new accessibility feature, the ability to block out Gravity Generator and Telecomms noises.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
I wonder if it wouldn't be better to unfuck those sounds. telecomms in particular seems mostly bad because of how much overlap it does, rather then anything else

**Comment by SuperNovaa41** [comment]
> I wonder if it wouldn't be better to unfuck those sounds. telecomms in particular seems mostly bad because of how much overlap it does, rather then anything else

It would probably be good just to do both tbh

**Review by Mothblocks — CHANGES_REQUESTED** [review]
:-1: to a few things here:

1. I don't think we should have an accessibility preference category. Accessibility features are often times just nice features that abled people find nice to use as well. They should just be under "gameplay" (the decision to put dark flashes there was a very intentional choice by me)
2. A loud sound setting is fine, but it should not be a variable on sounds. Instead, it should be a "max volume" slider or similar, and we clamp `volume` down every playsound/playsound_local/whatever.
3. Just lower the volume of the telecomms servers and gravity generator globally (though the preference as a global is fine). Their volumes were decided by someone who was literally hard of hearing!

**Comment by LemonInTheDark** [comment]
Before you lower the volume, try making the telecomms noise play from only once source
overlapping audio sounds really bad, specially in byond, and they do it by default

**Comment by JohnFulpWillard** [comment]
My main reason for wanting an Accessibility tab is that it should be the first thing you see when looking in settings. A lot of games I see with accessibility options like this always have it at the forefront and clearly labelled to know what it's meant for.
I also know an epileptic person who wasn't even aware that the flash darkness was even a thing until I told them myself, so the current place it's at doesn't seem to work for what the feature was supposed to do.

**Comment by LemonInTheDark** [comment]
> I'm also planning on adding more accessibility features in the future so it doesn't stick out oddly as much. Currently I had an idea for making screenshakes (during bombs) not so shake-y since Helianthus' PR lowering it wasn't enough and is still a problem.

shake_camera has a define that's "tiles per second" (really should be seconds per tile). You could make that a preference. It was something I was considering doing when I made my initial pr to smooth things out, but preferences weren't really done at the time

Would need to bound it of course, but even then a wide range looks more then fine

File: code/datums/looping_sounds/_looping_sound.dm [hunk 1]
```diff
@@ -37,16 +37,16 @@
 	var/skip_starting_sounds = FALSE
 	var/loop_started = FALSE
 	///Whether this looping sound takes character's sound accessibility preferences into account
-	var/player_sensitive = FALSE
+	var/loud_machine = FALSE
```
Comment (Mothblocks) [inline code/datums/looping_sounds/_looping_sound.dm:40]: Again, this should NOT be a per-sound thing. It should be clamped on volume itself.

Comment (JohnFulpWillard) [reply code/datums/looping_sounds/_looping_sound.dm:40]: that wound then affect all machinery, and would make this option less of an accessibility issue for those who have problems dealing with constant loud noise, and just another volume button.

Comment (Mothblocks) [reply code/datums/looping_sounds/_looping_sound.dm:40]: The point is that there's no realistic way for us to maintain what counts as an inaccessible loud sound and what doesn't, so we have to clamp all volume. Not LOWER all volume, CLAMP the volume.

**Pull request closed without merging**